sklearn/neighbors/_base.py

```diff
@@ -619,6 +618,22 @@ def _fit(self, X, y=None):
                 else:
                     self._fit_method = "brute"
 
+        if (
+            self.effective_metric_ in ["wminkowski", "minkowski"]
+            and self.effective_metric_params_["p"] < 1
+        ):
+            # For p < 1 minkowski is not a valid metric as it does not
+            # satisfy triangular inequality, but that is workable for
+            # bruteforce algorithm. But kd_tree and ball_tree requires
+            # valid metric to work
```

Comment:
Some rewording to be a more precise.
```suggestion
            # For 0 < p < 1 Minkowski distances aren't valid distance
            # metric as they do not satisfy triangular inequality:
            # they are semimetrics.
            # algo="kd_tree" and algo="ball_tree" can't be used because
            # KDTree and BallTree require distance metric to work properly.
            # However, the brute-force algorithm supports semimetrics.
```